HW06.Rmd

```diff
@@ -77,12 +98,39 @@ Now create a graphing function to create boxplots to visualize the results. Plot
 
 ```{r create a graphing function}
 #to work in the tidyverse, it will be easier to make tidy the dataframe first
+#Create a data set of values from H3K27_Mutated group only and reshape the data set from wide to long
+H3K27_Mutated <- GBM.genes.of.interest %>%
+  dplyr::select(gene_symbol, TPM...4:TPM...114) %>%
+  pivot_longer(cols = TPM...4:TPM...114) %>%
+  dplyr::select(gene_symbol, TPM = value) %>%
+  mutate(Group = "H3K27_Mutated")
+
+#Create a data set of values from WT group only and reshape the data set from wide to long
+WT <- GBM.genes.of.interest %>%
+  dplyr::select(gene_symbol, TPM...119:TPM...224) %>%
+  pivot_longer(cols = TPM...119:TPM...224) %>%
+  dplyr::select(gene_symbol, TPM = value) %>%
+  mutate(Group = "WT")
+
+#Combine two data sets together
+GBM.genes.of.interest_2 <- bind_rows(H3K27_Mutated, WT) %>%
+  arrange(gene_symbol, Group)
+
+#Create a graphing function for boxplot
+boxplot <- function(data) {
+ggplot(data, aes(Group, TPM, fill = Group)) +
+  geom_boxplot() + 
+  labs(title = paste(data$gene_symbol[1], "Expression in GBM Models by \n H3K27 Mutated or WT Status"), x = "H3K27\n", y = "\nExpression_(TPM)") +
+  theme(legend.position = "none", plot.title = element_text(hjust = 0.5))
+}
+

```

Comment:
I would suggest using a different function name because boxplot() masks the boxplot() function in baseR.

Reply:
Oh, I didn't realize this! Thank you!

Reply:
That is good to know